src/helper/bit-tools/fill-tool.js

```diff
@@ -0,0 +1,59 @@
+import paper from '@scratch/paper';
+import {floodFill, floodFillAll} from '../bitmap';
+import {getRaster} from '../layer';
+
+/**
+ * Tool for drawing fills.
+ */
+class FillTool extends paper.Tool {
+    /**
+     * @param {!function} onUpdateImage A callback to call when the image visibly changes
+     */
+    constructor (onUpdateImage) {
+        super();
+        this.onUpdateImage = onUpdateImage;
+        
+        // We have to set these functions instead of just declaring them because
+        // paper.js tools hook up the listeners in the setter functions.
+        this.onMouseDown = this.handleMouseDown;
+        this.onMouseDrag = this.handleMouseDrag;
+        this.onMouseUp = this.handleMouseUp;
+
+        this.color = null;
+        this.changed = false;
+        this.active = false;
+    }
+    setColor (color) {
+        this.color = color ? color : 'rgba(0,0,0,0)';
```

Comment:
Seems like we should be importing the default color string from somewhere (even if it's in a constant in this file) rather than hard-coding it.

Reply:
I did this and merged to head